econml/iv/dr/_dr.py

```diff
@@ -73,83 +246,1246 @@ def _effect_estimate(self, nuisances):
         cov_sign[cov_sign == 0] = 1
         clipped_cov = cov_sign * np.clip(np.abs(cov),
                                          self._cov_clip, np.inf)
-        return prel_theta + (res_y - prel_theta * res_t) * res_z / clipped_cov, clipped_cov
+        return prel_theta + (res_y - prel_theta * res_t) * res_z / clipped_cov, clipped_cov, res_z
+
+    def _transform_X(self, X, n=1, fitting=True):
```

Comment:
IT might be better that the final model mirrors more the structure in DRLearner. i.e.
1) the bias is not part of coef and we do not append a column of ones to X
2) we allow for an intercept in the final linear model in LinearDRIV and SparseLinearDRIV
3) we set the fit_cate_intercept to the fit_intercept of those models
4) we have the method "bias_part_of_coef" return False
5) we allow for X=None to be passed in the final model. Note that StatsModelsLinearRegression allows for X=None. So if someone just wants a constant model to be fitted, they can call LinearDRIV
6) This way we don't penalize the intercept in SparseLinearDRIV. Also if someone passes LassoCV as the final model, they don't penalize the intercept.

Reply:
As we discussed, since DRIV also handles continuous treatment, we can't mirror DRLearner structure, I will close all the related comments.